azure-pipelines/main-pipeline.yml

```diff
@@ -0,0 +1,228 @@
+
+trigger:
+
+  branches:
+    include: 
+    - experimental-pipeline
+
+pr:
+  branches:
+    include:
+    - main
+    - staging
+    - development
+    - feature_*
+    
+
+# Parameters and Variables to help with naming and calling variables, as well as providing a group of objects to loop over in later jobs.
+variables:
+    
+    org: '1054-daniel-felleman-revature-swag-shop'
+    projectKey: '1054-Daniel-Felleman-Revature-Swag-Shop_RevatureShopBackend'
+    projectName: 'RevatureShopBackend'
+
+parameters:
+- name: serviceConditions
+  type: object
+  default:
+  - accounts:
+    name: 'accounts'
+    requiresBase: true
+    jobName: 'AccountsBuild'
+  - commerce:
+    name: 'commerce'
+    requiresBase: true
+    jobName: 'CommerceBuild'
+  - config:
+    name: 'config'
+    jobName: 'ConfigBuild'
+  - eureka-server:
+    name: 'eureka-server'
+    jobName: 'EurekaBuild'
+  - inventory:
+    name: 'inventory'
+    requiresBase: true
+    jobName: 'InventoryBuild'
+  - Spring-Cloud-Gateway:
+    name: 'Spring-Cloud-Gateway'
+    jobName: 'SpringBuild'
+
+stages:
+- stage: Check
+  displayName: Check Service Changes, Test and Build
+  jobs:
+  - job: CheckChanges
+    pool: java-16
+    steps:
+
+# This script will check the Repo for changes and store them in a variable updatedFiles.
+# The script will iterate over every object stored in updatedFiles to select paths with changes.
+# Pipeline variable will be updated from 'false' to 'True' if a change was made in that path.
+# This step controls which build tasks will be triggered.
+# If Base is to be rebuilt, it will then set those services for which it is a dependency to build as well.   
+
+    - powershell: |
+        $updatedFiles = git diff HEAD HEAD~ --name-only
+          $updatedFiles | ForEach-Object {
+              Switch -Wildcard ($_ ) {
+                  'accounts/*' { Write-Output '##vso[task.setvariable variable=accounts;isOutput=true]True' }
+                  'base/*' { 
+                      Write-Output '##vso[task.setvariable variable=base;isOutput=true]True' 
+                      Write-Output '##vso[task.setvariable variable=accounts;isOutput=true]True'
+                      Write-Output '##vso[task.setvariable variable=commerce;isOutput=true]True'
+                      Write-Output '##vso[task.setvariable variable=inventory;isOutput=true]True'
+                  }
+                  'commerce/*' { Write-Output '##vso[task.setvariable variable=commerce;isOutput=true]True' }
+                  'config/*' { Write-Output '##vso[task.setvariable variable=config;isOutput=true]True' }
+                  'eureka-server/*' { Write-Output '##vso[task.setvariable variable=eureka-server;isOutput=true]True' }
+                  'inventory/*' { Write-Output '##vso[task.setvariable variable=inventory;isOutput=true]True' }
+                  'Spring-Cloud-Gateway/*' { Write-Output '##vso[task.setvariable variable=Spring-Cloud-Gateway;isOutput=true]True' }
```

Comment:
there's repetition here we could probably extract into a function, but ok

Reply:
That's a fair point. I like the readability of this. I might put this in a future release, but there are definitely more high priority work items.